### ISSUE TITLE
`node:crypto` `crypto.createCipheriv("aes-128-ecb", key, iv)` behaves differently from node

### ISSUE THREAD

**Issue by ZYinMD** [denoland/deno#22053]
Version: Deno 1.39.4

The following code involving `node:crypto` behaves differently from node:
```
$ node test.js
good!

$ deno run test.js
bad!
``` 
Possibly related: #18412 @kt3k 

Possible culprit is the `iv` argument.

```js
// test.js

import crypto from "node:crypto";
import { Buffer } from "node:buffer"; // this line is optional when using node

const key = "1234567890ABCDEF";

const iv = ""; // node outputs "good!", deno outputs "bad!"

// const iv = null; // node outputs "good!", deno throws

// const iv = new Uint8Array(16); // node throws, deno outputs "bad!"

export function encrypt(plainText) {
  const cipher = crypto.createCipheriv("aes-128-ecb", key, iv);
  return Buffer.concat([
    cipher.update(plainText),
    cipher.final(),
  ]).toString("base64");
}

export function decrypt(cipherText) {
  const cipher = crypto.createDecipheriv("aes-128-ecb", key, iv);
  return Buffer.concat([
    cipher.update(Buffer.from(cipherText, "base64")),
    cipher.final(),
  ]).toString("utf8");
}

const original = "hello world!";
const encrypted = encrypt(original); // "uMslN/bVLTsD2u4iu8v2fA=="
const decrypted = decrypt(encrypted); // "hello world!"

console.log(original === decrypted ? "good!" : "bad!");
```

**Comment by littledivy** [comment]
Ref https://github.com/denoland/deno/issues/22988

**Comment by kt3k** [comment]
It seems working in the latest Deno. The given example prints `good!` now. (Looks like it's fixed at v2.0.0. The last version which prints `bad!` seems like Deno 1.46.3)

**Comment by kt3k** [comment]
Seems like this was the same issue as #25279 and was fixed by #25571

```shellsession
$ deno upgrade --canary --version aae3a6bcb41fdb305d61a95f176be0db1513c56e
$ deno run a.mjs                                                          
good!
$ deno upgrade --canary --version 200145a09a51e96298f1ecd5ce78f75c27c7880a
$ deno run a.mjs                                                          
bad!
```
